astrokat/noisediode.py

```diff
@@ -9,257 +9,419 @@
     from katcorelib import user_logger
 except ImportError:
     from .simulate import user_logger
+from . import _DEFAULT_LEAD_TIME
+from . import max_cycle_len
 
 
-_DEFAULT_LEAD_TIME = 5.0  # lead time [sec]
+def _get_max_cycle_len(kat):
+    if not kat.dry_run:
+        return max_cycle_len(kat.sensor.sub_band.get_value())
+    else:
+        return max_cycle_len('l')
+
+
+def _eval_lead_time_(lead_time):
+    # if not given, apply default value
+    lead_time = float(lead_time or _DEFAULT_LEAD_TIME)
+    user_logger.trace('TRACE: Add lead time {}s to ND'
+                      .format(lead_time))
+    return lead_time
+
+
+def _set_time_(lead_time):
```

Comment:
I can understand your reasoning.
Since the functionality is to get the timestamp to send with the noise diode command, I am going to use the obvious name `_get_nd_timestamp_`